command/agent.go

```diff
@@ -434,6 +428,24 @@ func (c *AgentCommand) Run(args []string) int {
 
 			listeners = append(listeners, ln)
 
+			// Parse 'require_request_header' listener config option
+			var requireRequestHeader bool
+			if v, ok := lnConfig.Config[agentConfig.RequireRequestHeader]; ok {
+				switch v {
+				case true:
+					requireRequestHeader = true
+				case false /* noop */ :
+				default:
+					c.UI.Error(fmt.Sprintf("Invalid value for 'require_request_header': %v", v))
+					return 1
+				}
+			}
+
+			// Create a muxer and add paths relevant for the lease cache layer
+			mux := http.NewServeMux()
+			mux.Handle(consts.AgentPathCacheClear, leaseCache.HandleCacheClear(ctx))
+			mux.Handle("/", cache.Handler(ctx, cacheLogger, leaseCache, inmemSink, requireRequestHeader))
```

Comment:
Since the request header check is independent of the other functionality in cache.Handler what if we created a http handler that wraps the cache.Handler. e.g. something like: 
```
verifyRequestHeader(cache.Handler(ctx, cacheLogger, leaseCache, inmemSink))
```
Then we can choose to wrap the handler only if `lnConfig.Config[agentConfig.RequireRequestHeader]` is true.

Reply:
Good idea.  I coded that up, its definitely cleaner now.